src/compose.js

```diff
@@ -22,11 +19,18 @@ function createFactory(descriptor) {
     if (!descriptor.initializers || descriptor.initializers.length === 0) return obj;
 
     if (options === undefined) options = {};
-    return descriptor.initializers.filter(isFunction).reduce((resultingObj, initializer) => {
-      const returnedValue = initializer.call(resultingObj, options,
-        {instance: resultingObj, stamp: Stamp, args: [options].concat(args)});
-      return returnedValue === undefined ? resultingObj : returnedValue;
-    }, obj);
+    const inits = descriptor.initializers;
```

Comment:
This is a performance optimization. Removed unnecessary closure object creation.